lab-judy/test/person-route-test.js

```diff
@@ -0,0 +1,234 @@
+    it('should return 400 if no id was provided in request', function(done){
+      request.get(`${url}/api/person`)
+      .end((err, res) => {
+        expect(res.status).to.equal(404);
+        done();
+      });
+    });
+  });
+
+  describe('testing GET for request made with valid id', function(){
+    // * `GET` - test 200, response body like `{<data>}` for a request made with a valid id
+    before(done => {
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch( err => done(err));
+    });
+    //clean up the mocked person
+    after(done => {
+      if (this.tempPerson){
+        Person.deletePerson(this.tempPerson.id)
+        .then(() => done())
+        .catch(err => done(err));
+        return;
+      }
+      done();
+    });
+
+    it('should return 200 for a request with a valid id', done => {
+      request.get(`${url}/api/person/${this.tempPerson.id}`)
+      .end((err, res) => {
+        if (err) return done(err);
+        expect(res.status).to.equal(200);
+        expect(res.body.name).to.equal(examplePerson.name);
+        expect(res.body.age).to.equal(examplePerson.age);
+        done();
+      });
+    });
+  });
+
+
+  // *TODO:  `PUT` - test 400, responds with 'bad request' for if no `body provided` or `invalid body`
+  describe('testing PUT if no body provided or invalid body provided', function(){
+    before(done => {
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch( err => done(err));
+    });
+    //clean up the mocked person
+    after(done => {
+      Person.deletePerson(this.tempPerson.id)
+      .then(() => done())
+      .catch(err => done(err));
+    });
+    it('should return status 400 for no body provided or invalid body provided', done => {
+      request.put(`${url}/api/person/${this.tempPerson.id}`)
+      .set('Content-Type','application/json')
+      .send('invalid body')
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        done();
+      });
+    });
+  });
+
+  // * `PUT` - test 200, response body like  `{<data>}` for a post request with a valid body
+  describe('testing PUT for put request with valid body', function(){
+    before(done => {
+      Person.createPerson(examplePerson)
+      .then(person => {
+        this.tempPerson = person;
+        done();
+      })
+      .catch( err => done(err));
+    });
+    //clean up the mocked person
+    after(done => {
+      Person.deletePerson(this.tempPerson.id)
+      .then(() => done())
+      .catch(err => done(err));
+    });
+
+    it('should return 200 for request with valid body', done => {
+      let updatedPerson = {name: 'updated', age: '45'};
+      request.put(`${url}/api/person/${this.tempPerson.id}`)
+      .send(updatedPerson)
+      .end((err, res) => {
```

Comment:
You should do error handling in a test where you expect everything to go well (200):
```
if (err) return next(err);
```